### ISSUE TITLE
Argument type is unknown inside sum

### ISSUE THREAD

**Issue by mgmira** [microsoft/pyright#6992]
**Describe the bug**
Latest version of Pyright reports this error
```
/code/.debug/foo/bar.py
  /code/.debug/foo/bar.py:10:13 - error: Argument type is partially unknown
    Argument corresponds to parameter "__iterable" in function "sum"
    Argument type is "Generator[Unknown, None, None]" (reportUnknownArgumentType)
```

This error wasn't reported on v1.1.341

**Code or Screenshots**

```python
# /code/.debug/foo/bar.py
# pyright: strict

from __future__ import annotations


def get_value(x: int) -> int | None:
    ...


value = sum(max(value for x in range(10) if (value := get_value(x)) is not None) for _ in range(10))
```

**VS Code extension or command-line**
Pyright: [1.1.346](https://github.com/microsoft/pyright/releases/tag/1.1.346)

**Comment by erictraut** [comment]
Thanks for the bug report. I'm able to repro the problem as reported.

So far, I've narrowed it down to [this commit](https://github.com/microsoft/pyright/commit/27ff8c8dfb2bcad920675911bb678ad9218dc10d). I'll investigate further.

**Comment by erictraut** [comment]
This was a subtle long-standing bug that was exposed by a recent (unrelated) change. It will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.348, which was just published. It will also be included in a future release of pylance.
